File: android-app/app/src/main/java/com/inputblocker/app/ShutdownReceiver.java
package com.inputblocker.app;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

public class ShutdownReceiver extends BroadcastReceiver {
    
    private static final String TAG = "InputBlocker-Shutdown";
    
    @Override
    public void onReceive(Context context, Intent intent) {
        if (Intent.ACTION_SHUTDOWN.equals(intent.getAction())) {
            Log.i(TAG, "Shutdown detected");
            InputBlockerServiceManager.onShutdown();
        }
    }
}

File: android-app/app/src/main/java/com/inputblocker/app/ShutdownReceiver.kt
package com.inputblocker.app

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

class ShutdownReceiver : BroadcastReceiver() {
    
    companion object {
        private const val TAG = "InputBlocker-Shutdown"
    }
    
    override fun onReceive(context: Context, intent: Intent) {
        if (Intent.ACTION_SHUTDOWN == intent.action) {
            Log.i(TAG, "Shutdown detected")
            InputBlockerServiceManager.onShutdown()
        }
    }
}
